### PR TITLE
Tooltip updates

### PR DESCRIPTION
**Pull request by GomezIvann**
**Checklist**

- [x] Build process is done without errors. All tests pass in the `/lib` directory.
- [x] Self-reviewed the code before submitting.
- [x] Meets accessibility standards.
- [x] Added/updated documentation to `/website` as needed.
- [x] Added/updated tests as needed.

**Description**
Summary:

- Added a tooltip to long options with ellipses in the `Contextual Menu`.
- A tooltip was added to the `Select` component. Now our tooltip appears in long option labels and long selected values
- Updated tooltip implementation in our internal components. Now our tooltip doesn't add an extra container inside of our components.The build
- Added new title prop to the `Dropdown` component.
- Added new storybook tests to check all this functionality.

### CODE REVIEW

File: packages/lib/src/select/Select.stories.tsx
Comment (Mil4n0r) [inline packages/lib/src/select/Select.stories.tsx:1]: Multiple opinionated stories have been removed. I am unsure if this is a desired change.

**Comment by Mil4n0r** [comment]
I have been unable to find the reason why certain tooltips are not being displayed in the Chromatic's snapshots meanwhile they are visible in local storybook...

File: packages/lib/src/select/ListOption.tsx [hunk 1]
```diff
@@ -2,6 +2,9 @@ import styled from "styled-components";
 import { OptionProps } from "./types";
 import DxcCheckbox from "../checkbox/Checkbox";
 import DxcIcon from "../icon/Icon";
+import { useState } from "react";
+import { Tooltip } from "@radix-ui/react-tooltip";
```
[the remark this answers is not in the document]
Comment (GomezIvann) [reply packages/lib/src/select/ListOption.tsx:6]: Removed!

File: packages/lib/src/select/Select.stories.tsx
Comment (GomezIvann) [reply packages/lib/src/select/Select.stories.tsx:1]: I've added an especial story to test the opinionated theme!

File: packages/lib/src/tooltip/Tooltip.tsx [hunk 2]
```diff
@@ -120,14 +76,71 @@ const StyledTooltipContent = styled(Tooltip.Content)`
...
+        </Provider>
+      ) : (
+        children
+      )}
+    </TooltipContext.Provider>
+  );
+};
+
+export const TooltipWrapper = ({ condition, children, label }: TooltipWrapperProps) =>
+  condition ? <Tooltip label={label}>{children}</Tooltip> : <>{children}</>;
```
[the remark this answers is not in the document]
Comment (GomezIvann) [reply packages/lib/src/tooltip/Tooltip.tsx:142]: We have decided to keep the wrapper and use the regular Tooltip in the majority of components. Updated implementation!

**Review by Mil4n0r — APPROVED** [review]

**Pull request merged**